src/snowflake/snowpark/_internal/utils.py

```diff
@@ -525,27 +525,47 @@ def warning(name: str, text: str, warning_times: int = 1) -> None:
     warning_dict[name].warning(text)
 
 
-def deprecated(
-    *, deprecate_version: str, extra_warning_text: str = "", extra_doc_string: str = ""
+def func_decorator(
+    decorator_type: str, *, version: str, extra_warning_text: str, extra_doc_string: str
 ) -> Callable:
-    def deprecate_wrapper(func):
+    def wrapper(func):
         warning_text = (
-            f"{func.__name__} is deprecated since {deprecate_version}. "
+            f"{func.__qualname__}() is {decorator_type} since {version}. "
```

Comment:
We can use `__qualname__` to get a "fully-qualified" name in Python so we don't need to write it manually in decorator.